### DISCUSSION TITLE
Getting a file from a record in Go

### DISCUSSION THREAD

**Discussion by ndarilek** [pocketbase/pocketbase#4458] [Q&A] [answered]
Hey folks,

Working on an app using Pocketbase as a framework. I need to get a field on a record, "file", as either a file or its bytes.

How do I access file fields on records? I don't necessarily need an end-to-end example, but maybe something added to [this page](https://pocketbase.io/docs/go-records/) might help?

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
The record file operations will be simplified and documented better in the future, but for now here is one minimal example:

```go
key := record.BaseFilesPath() + "/" + record.GetString("yourFileField")

fsys, _ := app.NewFilesystem()
defer fsys.Close()

blob, _ := fsys.GetFile(key)
defer blob.Close()

// do something with the blob reader...
```

**Reply by danzilberdan** [reply]
Thanks. What if the files are stored on S3? Is that supported as well in this method?

**Reply by ganigeorgiev** [reply]
@danzilberdan Yes, `app.NewFilesystem()` in the above returns a filesystem interface based on the app storage settings, so if you have S3 enabled from the Admin UI, all operations with `fsys` will be against your S3 provider.

**Reply by Salman9000** [reply]
Hello, late to the party. But does pocketbase provide a function to get file as File object? In the documentation we can get url only. I am using nextjs.

**Reply by ganigeorgiev** [reply]
@Salman9000 It is not clear what actually you are asking, nor how it is related to the above discussion.

If you mean retrieving a file over the Web API - PocketBase provides a URL that you can download the file from - https://pocketbase.io/docs/api-files/#download-fetch-file. There is no helper is the JS SDK to generate a `File` object if that's what you are asking because I personally don't see the point of it. If for whatever reason you want to read the file in your Next.js server as blob you can simply send a plain `fetch` call to the PocketBase file url:

```js
await fetch(pb.files.getURL(...)).then((response) => response.blob())
```

If you need further help, feel free to open a new Q&A discussion but please try to elaborate a little more on the actual problem.

**Reply by Salman9000** [reply]
> There is no helper is the JS SDK to generate a File object if that's what you are asking.

Yes that is what i was asking, and yes this is exactly what i have implemented. Thanks.

**Comment by ndarilek** [comment]
Oh very helpful, thanks!
